### DISCUSSION TITLE
How can I use userInput in custom matcher?

### DISCUSSION THREAD

**Discussion by kobik** [zxcvbn-ts/zxcvbn#288] [Q&A]
Hi,

I'm trying to build a custom matcher following this [guide](https://zxcvbn-ts.github.io/zxcvbn/guide/matcher/#creating-a-custom-matcher). Is it possible to use in this custom matcher [userInput](https://zxcvbn-ts.github.io/zxcvbn/guide/user-input/)?

Thanks

**Comment by MrWook** [comment]
Hey, this is more of a discussion than a issue so i move it there.

But the userInput will be added to the global ZxcvbnOptions and is handled as every other dictionary.
So you can access it over the ZxcvbnOptions.
In version 3.x.x ZxcvbnOption is a singleton which is available globally and can be directly used in the matcher with an import.
So it could be accessed with:
```ts
import { zxcvbnOptions } from '@zxcvbn-ts/core'

const userInputDictionary = zxcvbnOptions.rankedDictionaries.userInputs
```
